packages/core/src/amazonqGumby/chat/controller/messenger/messenger.ts

```diff
@@ -0,0 +1,358 @@
+    JOB_SUBMISSION_STATUS_MESSAGE = 'gumbyJobSubmissionMessage',
+}
+
+export class Messenger {
+    public constructor(private readonly dispatcher: AppToWebViewMessageDispatcher) {}
+
+    public sendAnswer(params: { message?: string; type: ChatItemType; tabID: string; messageID?: string }) {
+        this.dispatcher.sendChatMessage(
+            new ChatMessage(
+                {
+                    message: params.message,
+                    messageType: params.type,
+                    messageId: params.messageID,
+                },
+                params.tabID
+            )
+        )
+    }
+
+    public sendErrorMessage(errorMessage: string, tabID: string) {
+        this.dispatcher.sendErrorMessage(
+            new ErrorMessage(`Sorry, we encountered a problem when processing your request.`, errorMessage, tabID)
+        )
+    }
+
+    public sendChatInputEnabled(tabID: string, enabled: boolean) {
+        this.dispatcher.sendChatInputEnabled(new ChatInputEnabledMessage(tabID, enabled))
+    }
+
+    public async sendAuthNeededExceptionMessage(credentialState: FeatureAuthState, tabID: string) {
+        let authType: AuthFollowUpType = 'full-auth'
+        let message = reauthenticateText
+        if (credentialState.amazonQ === 'disconnected') {
+            authType = 'full-auth'
+            message = reauthenticateText
+        }
+
+        if (credentialState.amazonQ === 'unsupported') {
+            authType = 'use-supported-auth'
+            message = enableQText
+        }
+
+        if (credentialState.amazonQ === 'expired') {
+            authType = 're-auth'
+            message = expiredText
+        }
+
+        this.dispatcher.sendAuthNeededExceptionMessage(new AuthNeededException(message, authType, tabID))
+    }
+
+    public sendAuthenticationUpdate(gumbyEnabled: boolean, authenticatingTabIDs: string[]) {
+        this.dispatcher.sendAuthenticationUpdate(new AuthenticationUpdateMessage(gumbyEnabled, authenticatingTabIDs))
+    }
+
+    public async sendProjectPrompt(modules: TransformationCandidateProject[], tabID: string) {
+        const moduleFormOptions: { value: string; label: string }[] = []
+        const uniqueJavaOptions = new Set<JDKVersion>()
+
+        modules.forEach(candidateModule => {
+            moduleFormOptions.push({
+                value: candidateModule.path,
+                label: candidateModule.name,
+            })
+
+            if (candidateModule.JDKVersion !== undefined) {
+                uniqueJavaOptions.add(candidateModule.JDKVersion)
+            }
+        })
+
+        const jdkFormOptions: { value: string; label: string }[] = []
+        uniqueJavaOptions.forEach(jdkVersion => {
+            jdkFormOptions.push({
+                value: jdkVersion.toString(),
+                label: jdkVersion.toString(),
+            })
+        })
+
+        const formItems: ChatItemFormItem[] = []
+        formItems.push({
+            id: 'GumbyTransformModuleForm',
```

Comment:
Syntax NIT: Put these ID's into string literal or enum.

Reply:
Good callout - this is something I wanted to do anyway, will do in the next revision